lib/plugins/amplitudePlugin.spec.ts

```diff
@@ -0,0 +1,134 @@
+import * as amplitude from '@amplitude/analytics-node'
+import type { BaseEvent, EnrichmentPlugin, Event } from '@amplitude/analytics-types'
+import { buildClient, sendGet } from '@lokalise/node-core'
+import type { FastifyInstance } from 'fastify'
+import fastify from 'fastify'
+
+import { amplitudePlugin, amplitudeTrack } from './amplitudePlugin'
+
+describe('amplitudePlugin', () => {
+  let app: FastifyInstance
+
+  beforeEach(() => {
+    app = fastify()
+  })
+
+  afterEach(async () => {
+    await app.close()
+  })
+
+  it('skip initialization if isEnabled is false', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+
+    await app.register(amplitudePlugin, {
+      isEnabled: false,
+    })
+
+    expect(initSpy).not.toHaveBeenCalled()
+  })
+
+  it('throw error if api key is not defined', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+
+    let error = null
+    try {
+      await app.register(amplitudePlugin, {
+        isEnabled: true,
+      })
+    } catch (e) {
+      error = e
+    }
+
+    expect(initSpy).not.toHaveBeenCalled()
+    expect(error).not.toBeNull()
+  })
+
+  it('Basic initialization', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+    const addHookSpy = jest.spyOn(app, 'addHook')
+
+    await app.register(amplitudePlugin, {
+      isEnabled: true,
+      apiKey: 'This is an api key',
+    })
+
+    expect(initSpy).toHaveBeenCalled()
+    expect(addHookSpy).toHaveBeenCalledTimes(0)
+  })
+
+  it('initialization tracking api usage', async () => {
```

Comment:
convention for test name is `(it) initializes tracking api`

However, basing tests on constructor assertions is fairly low-value, it says nothing about the behaviours that we actually care about. Instead it would be great to send a request and check that tracking actually happened

Reply:
You are right, we have another test for it so it is not useful, I will remove it.

Thanks for the test name convention 🙇